EC：デザインを変更してユーザー体験を向上させよう

KodaiMeboso opened the pull request:
### EC：デザインを変更してユーザー体験を向上させよう
**仕様**
次の画面仕様、機能仕様を実装する

**画面仕様**
![ECサイト：デザインを変更してユーザー体験を向上させよう](https://github.com/dotlife-dev/lxp-practical-project/assets/146696074/b6460cb8-74e1-493d-afb0-d63baea0941d)

**機能仕様**

- 画面レイアウトを画面仕様に従い修正する。
- 価格は日本円表示にする。１ドル140円で計算する。
- 価格の右横に送料を表示する。固定で980円。
- SKUを表示する。
- 「かごに追加」に用語を変更する。
- モバイルに対応するようレスポシンプルにする。

**方針**
- 価格は日本円表示にする。１ドル140円で計算する。
1. .envファイルを以下のように編集
```diff_ruby:
+ DEFAULT_CURRENCY=JPY
+ CURRENCY_SYMBOL=円
+ USD_TO_JPY_RATE=140
```
2. `laracom/project/config/cart.php`に以下を追加する。
```ruby
 'currency' => env('DEFAULT_CURRENCY', 'USD'),
 'currency_symbol' => env('CURRENCY_SYMBOL', '円'),
 'usd_to_jpy_rate' => env('USD_TO_JPY_RATE', 140),
 ```
3. `laracom/project/resources/views/layouts/front/product.blade.php`の
```ruby
{{ config('cart.currency') }} {{ $product->price }}
```
となっている箇所を
```ruby
{{ $product->price }}{{ config('cart.currency') }} 
```
に入れ替える。

- 価格の右横に送料を表示する。固定で980円。
`laracom/project/resources/views/layouts/front/product.blade.php`に以下を追加
```ruby
 +
 <span class="shipping-fee">送料980{{ config('cart.currency_symbol') }}</span>
```
「送料980円」はスタイルを後から変更できるようにspanタグで囲った

- SKUを表示する。
http://localhost:8000/admin/products/1/edit　にアクセスするとSKUが表示されていたので、該当するbladeファイルである
`laracom\project\resources\views\admin\products\edit.blade.php`からSKUのデータを参照している箇所を、
`laracom/project/resources/views/layouts/front/product.blade.php`に追加した。

**現在時点での画面**
![スクリーンショット 2024-01-25 124513](https://github.com/KodaiMeboso/lxp-practical-project/assets/146696074/503e3351-8a21-4b07-9a47-01ffd1c81589)


<!-- This is an auto-generated comment: release notes by coderabbit.ai -->
## Summary by CodeRabbit

- **新機能**
  - アプリに為替レート（USDからJPY）の設定機能を追加しました。
  - 商品価格、送料、SKU、数量入力フィールドの表示を改善しました。
<!-- end of auto-generated comment: release notes by coderabbit.ai -->

KodaiMeboso committed:
feat(config): change default currency to JPY and add conversion rate

This commit updates the default currency settings by changing from USD to JPY and also introduces a new environment variable `USD_TO_JPY_RATE` to handle conversions. Additionally, product price display logic has been updated to reflect the currency change and include shipping cost.

All text on product pages now reflects these changes with the inclusion of the SKU detail and translation of some placeholders and buttons for better localization in Japanese.

KodaiMeboso committed:
feat(currency): change default currency to USD and update display styling

The default currency configuration was updated from Japanese Yen (JYP) to US Dollar (USD) in the cart configuration file. Additionally, the product price and shipping fee display on the product page have been styled to be bold and the product price color has been changed to red with an increased font size. The 'Add to Cart' button width has also been adjusted for consistency. These changes are intended to reflect new business requirements for default currency and improve the visual emphasis on pricing information on the product pages.

KodaiMeboso commented on laracom/project/config/cart.php:
```diff
@@ -64,7 +64,7 @@
     ],
 
-    'currency' => env('DEFAULT_CURRENCY', 'JYP'),
+    'currency' => env('DEFAULT_CURRENCY', 'USD'),
```
ここの表示をドル表記に戻した理由は、管理者画面の商品編集画面で、商品の値段をUSDにするためです。JPYにしますと、赤枠の中がJPYになります。
管理者としてはドルで記入して、顧客が見る場所は円表示をするという意図があると思ったのでこのように編集しました！
![JPY](https://github.com/KodaiMeboso/lxp-practical-project/assets/146696074/2e4ecb96-bff1-461c-8129-5fe90607d335)

KodaiMeboso commented on laracom/project/resources/views/layouts/front/product.blade.php:
```diff
@@ -76,7 +78,7 @@ class="img-bordered img-responsive">
                             <input type="hidden" name="product" value="{{ $product->id }}" />
                         </div>
-                        <button type="submit" class="btn btn-warning"><i class="fa fa-cart-plus"></i>かごに追加</button>
+                        <button type="submit" class="btn btn-warning" style="width: 227px"><i class="fa fa-cart-plus"></i>かごに追加</button>
```
Buttonタぐのwidthを227pxに設定した理由は、Inputタグのwedthと揃えるためです。
chromeの拡張機能を使ってInputタグの幅を調べたら227pxでしたので、同じ幅にしました。
画像はPR作成時に後から撮影したのでなぜか226pxになっておりました。しかしButtonの幅も226pxで揃っています。

InputタグとButtonタグの幅を揃える方法があればご教授願います！
![width](https://github.com/KodaiMeboso/lxp-practical-project/assets/146696074/316bde99-28bc-4417-b893-731ddc1d82dc)

YuukiYamano replied:
@KodaiMeboso 
「数量」と「かごに追加」を常に同じ高さにするには２つのボタンを1 つのdivタグとかに入れてそのdivタグに対して高さを指定すると楽です！
同時に高さの問題226pxでも何故か揃ってしまうみたいな問題も解決できるかと！

KodaiMeboso reviewed (commented):
実装して気になった点が2点あるので、ご確認よろしくお願いいたします！

KodaiMeboso committed:
fix(frontend): align quantity field and add to cart button in product layout

Update the product detail layout on the front end to improve visual alignment of the quantity input field with the 'Add to Cart' button. Adjusted styles by wrapping both elements within a flex container, setting consistent widths and a margin to ensure proper spacing and alignment. This change enhances user interface consistency and improves the overall aesthetic of the product page.

KodaiMeboso commented on laracom/project/resources/views/layouts/front/product.blade.php:
```diff
@@ -73,12 +73,12 @@ class="img-bordered img-responsive">
-                            <input type="hidden" name="product" value="{{ $product->id }}" />
-                        </div>
-                        <button type="submit" class="btn btn-warning" style="width: 227px"><i class="fa fa-cart-plus"></i>かごに追加</button>
+                            <div class="form-group" style="display: flex;width: 464px;">
+                                <input type="text" class="form-control" name="quantity" id="quantity" style="margin-right: 1rem;width: 50%;"
+                                    placeholder="数量" value="{{ old('quantity') }}" />
+                                <input type="hidden" name="product" value="{{ $product->id }}" />
+                                <button type="submit" class="btn btn-warning" style="width: 50%;"><i class="fa fa-cart-plus"></i>かごに追加</button>
+                            </div>
```
Inputタグとbuttonタグをdivタグで囲いました！
するとInputタグとbuttonタグが垂直方向に並んだので、`display:flex`で水平方向に整列させました！
そしてdivタグの幅を464px取って、Inputタグとbuttonタグの幅を464pxの中で半分にしました！
464pxという数字は、先ほど設定した227px＊2≒460pxから、スタイルは8の倍数が好まれるという話から464pxに設定しました！

YuukiYamano replied:
OKですが、スタイルはcssファイルに記載してください！

KodaiMeboso replied:
`style.min.css`
が該当箇所のスタイルを司っているのは検証ツールでわかりました！
ただそのファイルがGit管理下にないからか、そのファイルの中身を変更しても反映されませんでした！
変更する場所に間違いがないかどうかは問題ありません、ただ編集ファイルが違うのかもしれませんね・・・引き続き調査して修正します！

KodaiMeboso reviewed (commented):
a7325748364ec69502f30f2e87edea77f77f3840
Inputタグとbuttonタグの幅を揃えたのでご確認よろしくお願いいたします！

KodaiMeboso committed:
style: improve formatting and input group display in front product view

This commit updates the formatting of mix-manifest.json for better readability. Additionally, it removes inline styling from the quantity input group in the front product layout, which improves maintainability and paves the way for more flexible CSS-based styling. The change to the form group's layout could impact visual presentation and should be tested for responsive behavior across devices.

KodaiMeboso committed:
style(front.css): enhance responsive design for footer and form elements

Updated the CSS to improve the responsive layout of the footer and form elements, ensuring better usability across devices with widths above 481px and below 480px. Adjusted the footer height, aligned form input elements with flex display, and standardized button and input sizes for mobile views. These changes aim to ensure a consistent and user-friendly interface regardless of device size.

KodaiMeboso committed:
style(front.css): rename .input to .form-control for consistency

Renamed CSS class `.input` to `.form-control` in `front.css` to maintain consistency with Bootstrap's convention. This change could affect any front-end components that reference the `.input` class, which should now be updated to use `.form-control`.

KodaiMeboso commented on laracom/project/resources/assets/css/front.css:
```diff
@@ -519,141 +519,319 @@ textarea, select {
+
+    #productAttribute {
+        width: 312px;
+    }
```
`@media screen and (max-width: 480px) {}`
`@media screen and (mix-width: 480px) {}`
をここに付与してレスポンシブ対応をしました！
パソコン版は以下のようになっています。
![スクリーンショット 2024-01-30 164646](https://github.com/KodaiMeboso/lxp-practical-project/assets/146696074/e8c313f9-2c89-4bfe-b37d-71678de57f39)

スマホ版は以下のようになっています。
変更前
![スクリーンショット 2024-01-30 144311](https://github.com/KodaiMeboso/lxp-practical-project/assets/146696074/c5182f5a-7e94-4aab-b615-bf09d015e79d)
変更後
![スクリーンショット 2024-01-30 165548](https://github.com/KodaiMeboso/lxp-practical-project/assets/146696074/aaa8f195-9b94-4cf9-838b-76dc65d3f018)

パソコン版でもスマホ版でも、SNSのアイコンが見えるようにしました。またスマホ版のテキストフィールドとカートボタンの長さをそろえました。

KodaiMeboso reviewed (commented):
レスポンシブ対応が完了したのでご確認よろしくお願いいたします。

KodaiMeboso commented on laracom/project/resources/assets/css/front.css:
```diff
@@ -684,7 +684,7 @@ textarea, select {
         width: 100%;
     }
-    .input {
+    .form-control {
```
inputクラスを指定しても、スマホ版におけるテキストフィールドとカートボタンの間に隙間が出なかったため、foam-controlクラスを指定することで解決しました。

KodaiMeboso reviewed (commented):
補足説明をしています。

YuukiYamano commented:
LGTM👍
